lib/gossip/dissemination.js

```diff
@@ -224,7 +216,74 @@ Dissemination.prototype._issueAs = function _issueAs(filterChange, mapChanges) {
         });
     }
 
-    return mapChanges(changesToDisseminate);
+    return changesToDisseminate;
+};
+
+Dissemination.prototype.tryStartReverseFullSync = function tryStartFullSync(target, timeout) {
+    var self = this;
+
+    if (this.reverseFullSyncJobs >= this.ringpop.maxReverseFullSyncJobs) {
+        this.logger.info('omit bidirectional full sync, too many already running', {
+            'remote': target,
+            'reverseFullSyncJobs': this.reverseFullSyncJobs,
+            'maxReverseFullSyncJobs': this.ringpop.maxReverseFullSyncJobs
+        });
+        this.ringpop.stat('increment', 'full-sync.reverse-omitted');
+        return;
+    }
+
+    this.reverseFullSyncJobs += 1;
+    this.ringpop.stat('gauge', 'full-sync.reverse-jobs', this.reverseFullSyncJobs);
+
+    var start = new Date();
+    this._reverseFullSync(target, timeout, onReverseFullSync);
+
+    function onReverseFullSync() {
+        self.ringpop.stat('timing', 'full-sync.reverse', start);
+        self.reverseFullSyncJobs -= 1;
+        self.ringpop.stat('gauge', 'full-sync.reverse-jobs', self.reverseFullSyncJobs);
+    }
+};
+
+Dissemination.prototype._reverseFullSync = function _reverseFullSync(target, timeout, callback) {
+    var self = this;
+    
+    this.ringpop.stat('increment', 'full-sync.reverse');
+    this.ringpop.client.protocolJoin({
+        host: target,
+        retryLimit: this.ringpop.config.get('tchannelRetryLimit'),
+        timeout: timeout
+    }, {
+        app: this.ringpop.app,
+        source: this.ringpop.whoami(),
+        incarnationNumber: this.ringpop.membership.localMember.incarnationNumber
+    }, onJoin);
+
+    function onJoin(err, res) {
+        if (err) {
+            self.logger.warn('bidirectional full sync failed due to failed join request', {
+                remote: target,
+                error: err
+            });
+            self.ringpop.stat('increment', 'full-sync.reverse.failed');
+            return callback();
+        }
+
+        process({
+            checksum: res.membershipChecksum,
+            members: res.membership
+        });
```

Comment:
I don't see you calling `callback` in non-error condition. If I am reading the code correctly, `self.reverseFullSyncJobs` will not get decremented if `protocolJoin` is successful.

Can you add a test which executes 6 reverse full syncs back-to-back, and verify they were all executed?